skhep/math/vectors.py

```diff
@@ -306,3 +308,236 @@ def __repr__(self):
     def __str__(self):
         """Simple class representation."""
         return str(tuple(self.__values))
+
+
+#-----------------------------------------------------------------------------
+# Lorentz vector class
+#-----------------------------------------------------------------------------
+class LorentzVector(object):
+    """
+    Class representing a Lorentz vector,
+    either a 4-dimensional Minkowski space-time vector or a 4-momentum vector.
+    The 4-vector components can be seen as (x,y,z,t) or (px,py,pz,E).
+
+    Constructors:
+        __init__(x=0., y=0., z=0., t=0.)
+        from4vector(avector)
+        from3vector(vector3D, t)
+    """
+    def __init__(self, x=0., y=0., z=0., t=0.):
```

Comment:
I was not necessarily thinking about a crazy user doing something rather silly (at least at 99% CL) as Vector3D(2,3) but rather about the neat possibility of just instantiating with "()". For sure there are situations where it's handy.